src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.cpp

```diff
@@ -518,30 +518,140 @@ ReturnCode_t TypeObjectRegistry::get_dependencies_from_type_object(
+                                {
+                                    break;
+                                }
+                            }
+                        }
+                    }
                     break;
                 case TK_SEQUENCE:
                     ret_code = get_sequence_array_dependencies(type_object.complete().sequence_type(),
                                     type_dependencies);
+                    //TODO Annotations are not currently supported, so their dependecies are ignored.
                     break;
                 case TK_ARRAY:
                     ret_code = get_sequence_array_dependencies(type_object.complete().array_type(), type_dependencies);
+                    //TODO Annotations are not currently supported, so their dependecies are ignored.
                     break;
                 case TK_MAP:
                     ret_code = get_map_dependencies(type_object.complete().map_type(), type_dependencies);
+                    //TODO Annotations are not currently supported, so their dependecies are ignored.
                     break;
                 // No dependencies
                 case TK_BITSET:
+                    if (ret_code == eprosima::fastdds::dds::RETCODE_OK &&
+                            type_object.complete().bitset_type().header().detail().ann_custom().has_value())
+                    {
+                        ret_code = get_custom_annotations_dependencies(
+                            type_object.complete().bitset_type().header().detail().ann_custom().value(),
+                            type_dependencies);
+                    }
+                    if (ret_code == eprosima::fastdds::dds::RETCODE_OK)
+                    {
+                        for (auto member : type_object.complete().bitset_type().field_seq())
+                        {
+                            if (member.detail().ann_custom().has_value())
+                            {
+                                ret_code = get_custom_annotations_dependencies(
+                                    member.detail().ann_custom().value(), type_dependencies);
+                                if (ret_code != eprosima::fastdds::dds::RETCODE_OK)
+                                {
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                    break;
                 case TK_ENUM:
+                    if (ret_code == eprosima::fastdds::dds::RETCODE_OK &&
+                            type_object.complete().enumerated_type().header().detail().ann_custom().has_value())
+                    {
+                        ret_code = get_custom_annotations_dependencies(
+                            type_object.complete().enumerated_type().header().detail().ann_custom().value(),
+                            type_dependencies);
+                    }
+                    break;
```

Comment:
CompleteEnumeratedLiteral also has a CompleteMemberDetail with custom annotations

Reply:
Added in [e6bbcb4](https://github.com/eProsima/Fast-DDS/pull/4497/commits/e6bbcb4eef25966bbcbe8431ae140705ef498cea)